File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/InvalidStringValueForDefaultTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class InvalidStringValueForDefaultTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().format().build());

    @Test
    public void testValidWhenPropertyIsSpecifiedPasses() throws ValidationException {
        // valid when property is specified
        final var schema = InvalidStringValueForDefault.InvalidStringValueForDefault1.getInstance();
        schema.validate(
            MapUtils.makeMap(
                new AbstractMap.SimpleEntry<String, String>(
                    "bar",
                    "good"
                )
            ),
            configuration
        );
    }

    @Test
    public void testStillValidWhenTheInvalidDefaultIsUsedPasses() throws ValidationException {
        // still valid when the invalid default is used
        final var schema = InvalidStringValueForDefault.InvalidStringValueForDefault1.getInstance();
        schema.validate(
            MapUtils.makeMap(
            ),
            configuration
        );
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/InvalidStringValueForDefaultTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class InvalidStringValueForDefaultTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun testValidWhenPropertyIsSpecifiedPasses() {
        // valid when property is specified
        val schema = InvalidStringValueForDefault.InvalidStringValueForDefault1.getInstance()
        schema.validate(
            mapOf(
                Pair(
                    "bar",
                    "good"
                )
            ),
            configuration
        )
    }

    @Test
    fun testStillValidWhenTheInvalidDefaultIsUsedPasses() {
        // still valid when the invalid default is used
        val schema = InvalidStringValueForDefault.InvalidStringValueForDefault1.getInstance()
        schema.validate(
            mapOf<Any?, Any?>(
            ),
            configuration
        )
    }
}
